python/ray/util/serialization_addons.py

```diff
@@ -11,31 +11,50 @@
 @DeveloperAPI
 def register_pydantic_serializer(serialization_context):
     try:
-        import pydantic.fields
+        from pydantic import fields
     except ImportError:
-        return
+        fields = None
 
-    # Pydantic's Cython validators are not serializable.
-    # https://github.com/cloudpipe/cloudpickle/issues/408
-    serialization_context._register_cloudpickle_serializer(
-        pydantic.fields.ModelField,
-        custom_serializer=lambda o: {
-            "name": o.name,
-            # outer_type_ is the original type for ModelFields,
-            # while type_ can be updated later with the nested type
-            # like int for List[int].
-            "type_": o.outer_type_,
-            "class_validators": o.class_validators,
-            "model_config": o.model_config,
-            "default": o.default,
-            "default_factory": o.default_factory,
-            "required": o.required,
-            "alias": o.alias,
-            "field_info": o.field_info,
-        },
-        custom_deserializer=lambda kwargs: pydantic.fields.ModelField(**kwargs),
+    try:
+        from pydantic.v1 import fields as pydantic_v1_fields
+    except ImportError:
+        pydantic_v1_fields = None
+
+    ModelField = (
+        fields.ModelField
+        if hasattr(fields, "ModelField")
+        else pydantic_v1_fields.ModelField
     )
 
+    if ModelField is not None:
```

Comment:
Q: Can't we just check if pydantic is 1 or 2 here instead?

Reply:
not sure why that's better. I thought it's better to check existance of features/variables over doing a version check.

Reply:
just the logic seems pretty complicated (1 vs 2 is just one line?) I am okay with this approach too.

Reply:
in this case, do you think it is possible to add a simple unit test?